File: android/src/main/java/com/lightningkite/kwift/observables/binding/ViewFlipper.binding.actual.kt
package com.lightningkite.kwift.observables.binding

import android.view.Gravity
import android.view.ViewGroup.LayoutParams.MATCH_PARENT
import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import android.view.animation.LinearInterpolator
import android.widget.CompoundButton
import android.widget.FrameLayout
import android.widget.ProgressBar
import android.widget.ViewFlipper
import com.lightningkite.kwift.observables.*

fun ViewFlipper.bindLoading(loading: MutableObservableProperty<Boolean>) {
    if (this.inAnimation == null)
        this.inAnimation = AlphaAnimation(0f, 1f).apply {
            duration = 250
            interpolator = LinearInterpolator()
        }
    if (this.outAnimation == null)
        this.outAnimation = AlphaAnimation(1f, 0f).apply {
            duration = 250
            interpolator = LinearInterpolator()
        }
    if (this.childCount == 1) {
        addView(ProgressBar(context), 1, FrameLayout.LayoutParams(WRAP_CONTENT, WRAP_CONTENT, Gravity.CENTER))
    }
    loading.addAndRunWeak(this) { self, it ->
        self.displayedChild = if (it) 1 else 0
    }
}

File: ios/Kwift/com/lightningkite/kwift/observables/binding/ViewFlipper.binding.actual.swift
//Stub file made with Kwift 2 (by Lightning Kite)
import Foundation
import UIKit


//--- ViewFlipper.bindLoading(MutableObservableProperty<Boolean>)
public extension ViewFlipper {
    func bindLoading(_ loading: MutableObservableProperty<Bool>) -> Void {
        if subviews.size == 1 {
            let new = UIActivityIndicatorView(frame: .zero)
            new.startAnimating()
            addSubview(new, FrameLayout.LayoutParams(gravity: .center))
        }
        loading.addAndRunWeak(self) { (self, value) in
            self.displayedChild = value ? 1 : 0
        }
    }
    func bindLoading(loading: MutableObservableProperty<Bool>) -> Void {
        return bindLoading(loading)
    }
}
















